apis/go.mod

```diff
@@ -1,14 +1,16 @@
 module github.com/metal3-io/baremetal-operator/apis
 
-go 1.22
+go 1.22.0
+
+toolchain go1.22.4
```

Comment:
We don't want `toolchain` directives, just put the required version right into `go` directive.

```suggestion
```

Reply:
It is still very controversial keyword, under discussion in Golang repo, prone to change more and it may affects ways of working and tooling we have (osv-scanner etc), so for now we did agree to stay off it as bumping to go directive achieves essentially the same.